testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -330,6 +335,91 @@ impl BlockMinerThread {
         Ok((aggregate_public_key, signature))
     }
 
+    /// Gather signatures from the signers for the block
+    fn gather_signatures(
+        &mut self,
+        new_block: &mut NakamotoBlock,
+        burn_block_height: u64,
+        stackerdbs: &mut StackerDBs,
+        attempts: &mut u64,
+    ) -> Result<(RewardSet, Vec<MessageSignature>), NakamotoNodeError> {
+        let Some(miner_privkey) = self.config.miner.mining_key else {
+            return Err(NakamotoNodeError::MinerConfigurationFailed(
+                "No mining key configured, cannot mine",
+            ));
+        };
+        let sort_db = SortitionDB::open(
+            &self.config.get_burn_db_file_path(),
+            true,
+            self.burnchain.pox_constants.clone(),
+        )
+        .expect("FATAL: could not open sortition DB");
```

Comment:
Please don't make these fatal.  Just return an error; the miner will try again on its own.

Reply:
This code was moved up into `run_miner` in https://github.com/stacks-network/stacks-core/pull/4807/commits/aa9baf977c5edfa6b28e47d7f9896e1f3a84c2ef, which is in #4807

Reply:
This is still fatal in this PR, now that #4807 has merged.  Please make it fallible.

Reply:
I believe this is addressed, but please take a look and lmk if otherwise. Thanks!